src/app/db/initDatabase.ts

```diff
@@ -26,27 +26,36 @@ import * as app from 'app/util/applib';
 
 const log = app.createLog('db:mongo');
 
-export default function () : Promise <void> {
-	const mongoURL = config.get('db.mongo.url');
-
-	const options : MoreConnectionOptions = {
+const mongoURL = config.get('db.mongo.url');
+const options: MoreConnectionOptions = {
 	//	autoIndex: !app.isProduction,  // Option not supported, although the docs mention it
-		connectTimeoutMS: 4000,
-		keepAlive: 120,
-		useMongoClient: true,
-		reconnectTries: Number.MAX_VALUE,
-		socketTimeoutMS: 2000,
-	};
+	connectTimeoutMS: 4000,
+	keepAlive: 120,
+	useMongoClient: true,
+	reconnectTries: Number.MAX_VALUE,
+	socketTimeoutMS: 2000,
+};
+const reconnectionLimit = 5;
+let reconnectionAmount = 1;
```

Comment:
The `let ...` declaration should be inside the `new Promise(... => {` arrow function. It should automatically create a closure. If this is a module-scoped variable, it will apply to all calls to the exported default function and could lead to concurrency issues.